src/components/storageSrv/storageFirebaseAdapterSrv.js

```diff
@@ -20,7 +20,7 @@
                     keys.forEach(function(key){
                         var value = source[key];
 
-                        if(key[0] === '$' || angular.isUndefined(value) || (angular.isArray(value) && !value.length)){
+                        if(key[0] === '$' || angular.isUndefined(value) || (angular.isArray(value) && !value.length) || (value !== value)){//value !== value return true if it equals to NaN
```

Comment:
1) checking if start with $.
2) this is the quickest way i found to check if a var equals to NaN, isNaN cannot be used since the following expression will be equals to true isNaN('test').